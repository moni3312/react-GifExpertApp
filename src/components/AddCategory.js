import { useState } from "react"
import PropTypes from "prop-types";

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) =>{
        e.preventDefault();//para que no recargue la pagina

        if(inputValue.trim().length > 2){
        setCategories(cats => [inputValue, ...cats]);
        setInputValue('');
        }
    }

    return (
        <form onSubmit= {handleSubmit}>
            <p>{inputValue}</p>
                <input 
                    type="text"
                    placeholder= "Escriba el tema de su Gif....."
                    value= {inputValue}
                    onChange= {handleInputChange}
                />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
