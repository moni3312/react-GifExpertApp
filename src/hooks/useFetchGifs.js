import { useEffect, useState } from "react"
import { getGIfs } from "../helpers/getGifs";

export const useFetchGifs = (category) => {
    
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(()=>{

        getGIfs(category)
            .then(imgs => {
                setState({
                    data: imgs,
                    loading: false
                })//setState
            })

    },[category]) 
//useEffect lo que hace es recargar el componente solo si cambia lo que esta entre [] osea solo si cambia category
//para que no afecte nigun boton o algo externo haga que se vuelva a cargar el componente o en este caso el getGifs

    return state; //{data:[], loading:true};
}