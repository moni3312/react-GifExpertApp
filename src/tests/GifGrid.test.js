import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import { GifGrid } from '../components/GifGrid';
import { useFetchGifs } from '../hooks/useFetchGifs';
jest.mock('../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />',()=> {
    const category = 'hola';
  

    test('deben de mostrar <GifGrid /> correctamente', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('deben de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }]
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>);
  
        expect( wrapper).toMatchSnapshot();
    });

});