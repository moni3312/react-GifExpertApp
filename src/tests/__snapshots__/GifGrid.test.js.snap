// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> deben de mostrar <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3
    className=" animate__animated animate__fadeIn"
  >
    hola
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading..
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> deben de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className=" animate__animated animate__fadeIn"
  >
    hola
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
